### PR TITLE
:seedling: Release-1.7 related changes

### LINKED ISSUE #1621: Add release-1.7 related tests and configuration in e2e
Now that we have releases v1.7.x we also need to add the corresponding tests and configuration for e2e.

One such test would be the clusterctl upgrade test here: https://github.com/metal3-io/cluster-api-provider-metal3/blob/main/test/e2e/upgrade_clusterctl_test.go#L30 

And also add v1.7.x related e2e configs here https://github.com/metal3-io/cluster-api-provider-metal3/blob/main/test/e2e/config/e2e_conf.yaml#L20 

Also add a documentation in the release doc so that we do it for future minor releases https://github.com/metal3-io/cluster-api-provider-metal3/blob/main/docs/releasing.md#post-release-actions-for-new-release-branches

### PR DESCRIPTION
**Pull request by peppi-lotta**
Added a test and configurations for release-1.7. Also some minor changes to comments and documentation.

**What this PR does / why we need it**:

**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*:
Fixes #1621

### CODE REVIEW

**Review by Sunnatillo — COMMENTED** [review]
/lgtm

File: docs/e2e-test.md [hunk 1]
```diff
@@ -101,6 +101,7 @@ in the ephemeral cluster either before pivoting or after re-pivoting.
 
 | tests         | CAPM3 from  | CAPM3 to  | CAPI from  | CAPI to         |
 | --------------| ----------- | --------- | ---------- |---------------- |
+| v1.7=>current | v1.7.1      | main      | v1.7.1     | latest release  |
```
Comment (adilGhaffarDev) [inline docs/e2e-test.md:104]:
```suggestion
| v1.7=>current | v1.7.0      | main      | v1.7.1     | latest release  |
```
v1.7.1 is not released yet. let's change it when we do the patch release.

**Pull request merged**